hoodie-client/src/main/java/com/uber/hoodie/io/HoodieIOHandle.java

```diff
@@ -104,4 +110,29 @@ public Schema getSchema() {
   public static Schema createHoodieWriteSchema(HoodieWriteConfig config) {
     return HoodieAvroUtils.addMetadataFields(new Schema.Parser().parse(config.getSchema()));
   }
+
+  // Dynamic calculation of maxMemory to use for merge
+  protected long getMaxMemoryAllowedForMerge() {
+
+    if (SparkEnv.get() != null) {
+      long executorMemoryInBytes = Long.valueOf(
+          Utils.memoryStringToMb(SparkEnv.get().conf().get(config.getSparkExecutorMemoryProp(),
+              config.getDefaultSparkExecutorMemory())) * 1024
+              * 1024L); // 1 GB is the default conf used by Spark, look at SparkContext.scala
```

Comment:
how do we intend to keep up with the default.. is there a equivalent way to do this dynamically as below?

Reply:
I checked earlier, it's hardcoded in Spark..:( 

https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L465